spotlight-client/src/VizHistoricalPopulationBreakdown/VizHistoricalPopulationBreakdown.tsx

```diff
@@ -15,25 +15,86 @@
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 // =============================================================================
 
-import { startOfMonth } from "date-fns";
+import { startOfMonth, sub } from "date-fns";
 import { observer } from "mobx-react-lite";
-import React from "react";
-import { WindowedTimeSeries } from "../charts";
+import { rem } from "polished";
+import React, { useState } from "react";
+import styled from "styled-components/macro";
+import { isWindowSizeId, WindowedTimeSeries, WindowSizeId } from "../charts";
 import type HistoricalPopulationBreakdownMetric from "../contentModels/HistoricalPopulationBreakdownMetric";
+import DemographicFilterSelect from "../DemographicFilterSelect";
+import { colors, Dropdown, zIndex } from "../UiLibrary";
+
+const FilterRow = styled.div`
+  background: ${colors.background};
+  display: flex;
+  justify-content: flex-end;
+  padding-bottom: ${rem(16)};
+  z-index: ${zIndex.control};
+`;
+
+const FilterWrapper = styled.div`
+  margin: 0 ${rem(16)};
+
+  &:first-child {
+    margin-left: 0;
+  }
+
+  &:last-child {
+    margin-right: 0;
+  }
+`;
 
 const VizHistoricalPopulationBreakdown: React.FC<{
   metric: HistoricalPopulationBreakdownMetric;
 }> = ({ metric }) => {
-  // TODO(#278): implement filter UI to change this
-  const defaultRangeEnd = startOfMonth(new Date());
+  const [windowSizeId, setWindowSizeId] = useState<WindowSizeId>("20");
+
+  let defaultRangeEnd = startOfMonth(new Date());
+  if (!metric.dataIncludesCurrentMonth) {
+    defaultRangeEnd = sub(defaultRangeEnd, { months: 1 });
+  }
+
+  let defaultRangeStart: Date | undefined;
+  if (windowSizeId !== "custom") {
+    defaultRangeStart = sub(defaultRangeEnd, {
+      years: Number(windowSizeId),
+      // make the range start-exclusive to correct for an off-by-one error
+      months: -1,
+    });
+  }
 
   if (metric.dataSeries)
     return (
-      <WindowedTimeSeries
-        data={metric.dataSeries}
-        setTimeRangeId={() => undefined}
-        defaultRangeEnd={defaultRangeEnd}
-      />
+      <>
+        <FilterRow>
+          <FilterWrapper>
+            <Dropdown
```

Comment:
I'm getting a warning from this Dropdown. When the page initially loads, when I choose a custom window size, I get

<img width="1773" alt="Screen Shot 2021-01-26 at 5 02 40 PM" src="https://user-images.githubusercontent.com/5872651/105928246-6f60a700-5ffa-11eb-9aae-d1e5bdc58564.png">

Reply:
yeah this is annoying but I'm pretty sure it's related to [this known issue](https://github.com/pmndrs/react-spring/issues/906) in the latest stable version of `react-spring` that supposedly will be fixed in the next major release. as far as I can tell there isn't really anything we can do about it in our code? not the first time I've seen it